### PR TITLE
AUT-1194 - Document Account Management API and include proposed changes

### PR DESCRIPTION
**Pull request by JHjava**
## What?

- Document the existing Account Management API along with the proposed 2 new API endpoints where one will retrieve the users MFA method and the other will allow the MFA method to be updated or changed. 
- Once the API endpoint to update the users MFA method is added, the update users phone number API endpoint can be removed 

## Why?

- To clearly document what the Account Management API offers
- To document the proposed changes and adaptations to the Account Management API

### CODE REVIEW

File: account-management-api/docs/account-management-api.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+* send a one-time passcode (OTP) notification to a user
+
+
+## Accessing the account management API
+
+The account management API is restricted to the account management RP using an API Gateway Lambda authoriser to control access.
+
+To use the API, the account management RP needs to:
+
+1. Get an access token from the /token endpoint - find more [information on how to request an access token](https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/integrate-with-code-flow/#make-a-post-request-to-the-token-endpoint).
```
Comment (benjamin-mortimer) [inline account-management-api/docs/account-management-api.md:21]:
```suggestion
1. Get an access token from the `/token` endpoint - find more [information on how to request an access token](https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/integrate-with-code-flow/#make-a-post-request-to-the-token-endpoint).
```

File: account-management-api/docs/account-management-api.md [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+| `primaryMfaMethod` | The user’s primary MFA method when signing in to [GOV.UK One Login services at medium Cl.Cm protection level](https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/choose-the-level-of-authentication/#choose-the-level-of-authentication-for-your-service). <br><br> The 2 available methods are: <ul><li>`AUTH_APP` for users wanting to use an authenticator app</li><li>`SMS` for users wanting to receive OTPs on their mobile phone</li></ul> |
+
+### Update a user’s MFA method
+
+You can make a `POST` request to the `/update-mfa-method` endpoint to change a user’s MFA method. 
+
+When changing the MFA method from authenticator app to SMS, you must call the [`/send-otp-notification` endpoint](https://github.com/alphagov/di-authentication-api/edit/bau-draft-account-management-api-doc/account-management-api/docs/account-management-api.md#send-an-otp-notification-to-a-user) first to send an OTP to the user's mobile phone.
+
+If successful,
+* we remove the existing MFA method
```
Comment (benjamin-mortimer) [inline account-management-api/docs/account-management-api.md:181]: Is there a missing bullet here? Presumably this request also adds the new MFA method?

File: account-management-api/docs/account-management-api.md [hunk 3]
```diff
@@ -0,0 +1,240 @@
+Status 16/05/2023
+
+# Understand the account management API
+
+The account management API is an internal interface used by the account management relying party (RP) to access the authentication database. It provides functionality to
```
Comment (benjamin-mortimer) [inline account-management-api/docs/account-management-api.md:5]:
```suggestion
The account management API is an internal interface used by the account management relying party (RP) to access the authentication database. It provides functionality to: 
```

**Comment by benjamin-mortimer** [comment]
This looks great @JHjava. 

I've picked out a couple of typos.

Otherwise, we'll try and get you a more substantive tech review in the near future - apologies for the delay on this.

**Review by dbes-gds — APPROVED** [review]

**Pull request merged**